### PR TITLE
fix(modal): ensure that v-model is updated when show or hide is canceled

### PR DESCRIPTION
**Pull request by tmorehouse**
### Describe the PR

Addresses https://github.com/bootstrap-vue/bootstrap-vue/issues/3126#issuecomment-484347044

Specifically: When the v-model is used to open or hide the modal, and the corresponding show or hide event was cancelled, the v-model was not being reset.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/modal/modal.js [hunk 1]
```diff
@@ -361,6 +361,11 @@ export default Vue.extend({
     }
   },
   methods: {
+    updateModel(val) {
+      if (val !== !!this.visible) {
```
Comment (jackmu95) [inline src/components/modal/modal.js]: Why the `!!`? The `visible` prop should always be a boolean.

Comment (tmorehouse) [reply src/components/modal/modal.js]: "Should" be is the keyword, except when someone uses a non Boolean on the prop (and has dev mode off which won't warn the user)

Comment (tmorehouse) [reply src/components/modal/modal.js]: The other option was to use:
```js
if (val === !this.visible)
```

Comment (jackmu95) [reply src/components/modal/modal.js]: Hmm, but we expect the type at most other places... But I'am also fine with using it.

Comment (tmorehouse) [reply src/components/modal/modal.js]: I suppose we should probably emit it regardless of type... that way the model is updated correctly

**Pull request merged**